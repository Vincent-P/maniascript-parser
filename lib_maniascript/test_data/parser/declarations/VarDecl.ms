declare Ident   G_InvisibleSkinId;

main()
{
    declare metadata Meta_IsPortal for _Landmark = False;
}
