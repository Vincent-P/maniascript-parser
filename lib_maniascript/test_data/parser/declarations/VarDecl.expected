NODE_ROOT 0..104 {
  NODE_VAR_DECL 0..34 {
    TOKEN_DECLARE("declare") 0..7
    TOKEN_WHITESPACE(" ") 7..8
    NODE_TYPE 8..13 {
      NODE_IDENTIFIER 8..13 {
        TOKEN_IDENT("Ident") 8..13
      }
    }
    TOKEN_WHITESPACE("   ") 13..16
    NODE_IDENTIFIER 16..33 {
      TOKEN_IDENT("G_InvisibleSkinId") 16..33
    }
    TOKEN_SEMICOLON(";") 33..34
  }
  TOKEN_WHITESPACE("\n\n") 34..36
  NODE_FUNC_DECL 36..104 {
    NODE_IDENTIFIER 36..40 {
      TOKEN_IDENT("main") 36..40
    }
    TOKEN_OPEN_PAREN("(") 40..41
    TOKEN_CLOSE_PAREN(")") 41..42
    TOKEN_WHITESPACE("\n") 42..43
    NODE_BLOCK 43..104 {
      TOKEN_OPEN_BRACE("{") 43..44
      TOKEN_WHITESPACE("\n    ") 44..49
      NODE_VAR_DECL 49..102 {
        TOKEN_DECLARE("declare") 49..56
        TOKEN_WHITESPACE(" ") 56..57
        TOKEN_METADATA("metadata") 57..65
        TOKEN_WHITESPACE(" ") 65..66
        NODE_IDENTIFIER 66..79 {
          TOKEN_IDENT("Meta_IsPortal") 66..79
        }
        TOKEN_WHITESPACE(" ") 79..80
        TOKEN_FOR("for") 80..83
        TOKEN_WHITESPACE(" ") 83..84
        NODE_EXPRESSION 84..93 {
          NODE_IDENTIFIER 84..93 {
            TOKEN_IDENT("_Landmark") 84..93
          }
        }
        TOKEN_WHITESPACE(" ") 93..94
        TOKEN_EQUAL("=") 94..95
        TOKEN_WHITESPACE(" ") 95..96
        NODE_EXPRESSION 96..101 {
          NODE_IDENTIFIER 96..101 {
            TOKEN_IDENT("False") 96..101
          }
        }
        TOKEN_SEMICOLON(";") 101..102
      }
      TOKEN_WHITESPACE("\n") 102..103
      TOKEN_CLOSE_BRACE("}") 103..104
    }
  }
}